index.js

```diff
@@ -111,6 +116,44 @@ Babel.prototype.optionsHash = function() {
       hash[key] = (typeof value === 'function') ? (value + '') : value;
     }
 
+    if (options.plugins) {
+      hash.plugins = [];
+
+      for (var i = 0; i < options.plugins.length; i++) {
+        var plugin = options.plugins[i];
+
+        var pluginType = typeof plugin;
+        var pluginAugmentsCacheKey = false;
+        var pluginProvidesBaseDir = false;
+
+        // do not look for properties on string plugins
+        if (pluginType === 'function' || pluginType === 'object') {
+          pluginAugmentsCacheKey = typeof plugin.cacheKey === 'function';
+          pluginProvidesBaseDir = typeof plugin.baseDir === 'function';
+        }
+
+        if (pluginAugmentsCacheKey) {
+          hash.plugins.push(plugin.cacheKey());
+        }
+
+        if (pluginProvidesBaseDir) {
+          var depHash = hashForDep(plugin.baseDir());
+
+          hash.plugins.push(depHash);
+        }
+
+        if (pluginType === 'string') {
+          hash.plugins.push(plugin);
+        } else if (!pluginProvidesBaseDir){
+          // prevent caching completely if the plugin doesn't provide baseDir
+          // we cannot ensure that we aren't causing invalid caching pain...
+          this.console.log('broccoli-babel-transpiler is opting out of caching due to a plugin that does not provide a caching strategy: `' + plugin + '`.');
+          hash.plugins.push((new Date).getTime() + Math.random());
```

Comment:
I will switch to make sure that the numbers done merge (by coercing to strings), but when I wrote it I thought it might be good actually. Because `getTime` always returns an integer, and `Math.random()` always returns a float from 0 to 1 we would just end up with a value like `1444566677.129387273`.